bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -127,6 +127,17 @@ private MergeSource(String repositoryName, String branchName) {
             if (!branchMatcher.matches()) {
                 return Optional.empty();
             }
+
+            // Verify that the branch exists
+            var isValidBranch = prInstance.remoteBranches().stream()
+                                          .map(Reference::name)
+                                          .anyMatch(branch -> branch.equals(branchMatcher.group(1)));
+            if (!isValidBranch) {
+                // Assume the name refers to a sibling repository
+                var repoName = Path.of(pr.repository().name()).resolveSibling(branchMatcher.group(1)).toString();
```

Comment:
Perhaps check that the repo actually exists?

Reply:
It will be checked a bit later (around line 193).